DPPP/BDABuffer.cc

```diff
@@ -82,7 +83,8 @@ namespace DP3 {
     , weights_(other.weights_)
     , full_res_flags_(other.full_res_flags_)
     , rows_()
-    , remaining_capacity_(0)
+    , original_capacity_(other.original_capacity_)
```

Comment:
The capacity of the copy will typically be smaller than the original: Copying a std::vector (the memory pool) does not copy its capacity: It only copies the contents. I'll have to check the behavior for ao::uvector, but my intention was that the copy has no remaining capacity.
You can use the loop over the rows, below, for initializing original_capacity_.

Reply:
I would expect that if you have a buffer that still has 12 pools left that a copy would also have 12 pools left. This is because I'd expect that calls on copies are similar to calls on the original one.

Calling AddRow on the original will succeed and on the copy will return false contradicts this.

However, this is my opinion on how copy/clone should work based on experience in Java. Could be that it is different in c++ in which case I would like to hear it as well and then we can fix it :)